.github/workflows/go_wasm_build_test.yml

```diff
@@ -0,0 +1,27 @@
+# This workflow will build a golang project
```

Comment:
We can make this a reusable workflow, so we dont have to copy  the same steps .github/workflows/go_wasm_deploy.yml

Reply:
Good idea. Unfortunately reusable workflows won't work as we still need to run `./build_wasm` again in the context of the deploy workflow.

I've instead consolidated all of it into a single action. And only in case of push events against `main`, will the deployment part run.